src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -255,7 +268,7 @@ abstract class AbstractCheckTask(
         try {
             val currentState = taskState.get()
             LOGGER.info("Finishes task '$description' in state $currentState")
-            if (currentState == State.TIMEOUT) {
+            if (currentState == State.TIMEOUT || currentState == State.MESSAGE_TIMEOUT || currentState == State.STREAM_COMPLETED) {
```

Comment:
I believe we can add the field into the enum that indicated whether the state should trigger `callOnTimeoutCallback` method call. And it will simplify the condition